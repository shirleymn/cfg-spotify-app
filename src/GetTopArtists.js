import React, { useEffect, useState } from "react";
import axios from "axios";

const TOP_ARTISTS_ENDPOINT = `https://api.spotify.com/v1/me/top/artists`



const GetTopArtists = () => {

    const [access_token, setToken] = useState("");
    const [data, setData] = useState({});

    useEffect(() => {
        if (localStorage.getItem('accessToken')) {
            setToken(localStorage.getItem("accessToken"));
        }
    }, []);

    const TopArtistsLongTerm = () => {
        axios
        .get(
            TOP_ARTISTS_ENDPOINT, {
                params: {limit: 10, offset: 0, time_range: 'long_term',},
            headers: {
                Accept: 'application/json',
                Authorization: "Bearer " + access_token,
                'content-type': 'application/json'
            }
        }).then(response => {
            setData(response.data);
        })
        .catch((error) => {
            console.log(error)
        });
    };

    const TopArtistsMediumTerm = () => {
        axios
        .get(
            TOP_ARTISTS_ENDPOINT, {
                params: {limit: 10, offset: 0, time_range: 'medium_term',},
            headers: {
                Accept: 'application/json',
                Authorization: "Bearer " + access_token,
                'content-type': 'application/json'
            }
        }).then(response => {
            setData(response.data);
        })
        .catch((error) => {
            console.log(error)
        });
    };

    const TopArtistsShortTerm = () => {
        axios
        .get(
            TOP_ARTISTS_ENDPOINT, {
                params: {limit: 10, offset: 0, time_range: 'short_term',},
            headers: {
                Accept: 'application/json',
                Authorization: "Bearer " + access_token,
                'content-type': 'application/json'
            }
        }).then(response => {
            setData(response.data);
        })
        .catch((error) => {
            console.log(error)
        });
    };

    

  

    return (
        <div>
            
        <button onClick={TopArtistsShortTerm}>Past 4 weeks top artists</button>   
        <button onClick={TopArtistsMediumTerm}>Past 6 months top artists</button>
        <button onClick={TopArtistsLongTerm}>Get all-time top artists</button>
 
        {data?.items ? data.items.map((item) => <p>{item.name}</p>) : null}
        
        </div>
    ) 
    

};



export default GetTopArtists