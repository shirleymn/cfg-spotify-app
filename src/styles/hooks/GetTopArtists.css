@import url('https://fonts.googleapis.com/css2?family=Inter:wght@500;700&display=swap');

p {
    font-size: 24px;
    font-family: 'Inter', sans-serif;
    font-weight: medium;
}

.container {
    display: grid;
    grid-template-columns: repeat(3, [col] 300px);
    grid-template-rows: repeat(20, [row] );
    grid-column-gap: 0px;
    grid-row-gap: 50px;
    /* background-color: rgb(177, 20, 20); */

}

.rank-column {
    /* background-color: #8e3eda; */
    justify-self: end;
    align-self: center;


}
.rank {
    font-size: 48px;
}

.image-column {
    /* background-color: #1898d4; */
    align-self: center;
    justify-self: center;

}

img {
    max-width: 50%;  
    height: auto;
    filter: drop-shadow(2px 2px 2px rgb(126, 126, 126));
}

.names-column {
    grid-column: col 3;
    display: grid;
    grid-row-gap: 0px;
    grid-template-rows: 50px 50px;
    /* background-color: #ce18d4; */
    justify-self: start;
    align-self: center;
}

