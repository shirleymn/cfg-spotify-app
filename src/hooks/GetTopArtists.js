import React, { useEffect, useState } from "react";
import axios from "axios";
import { Button } from "@material-ui/core";
import '../styles/hooks/GetTopArtists.css'


const TOP_ARTISTS_ENDPOINT = `https://api.spotify.com/v1/me/top/artists`



const GetTopArtists = () => {

    const [access_token, setToken] = useState("");
    const [data, setData] = useState({});

    useEffect(() => {
        if (localStorage.getItem('accessToken')) {
            setToken(localStorage.getItem("accessToken"));
        }
        console.log(data)
    }, []);

      const TopArtistsLongTerm = () => {
        axios
        .get(
            TOP_ARTISTS_ENDPOINT, {
                params: {limit: 10, offset: 0, time_range: 'long_term',},
            headers: {
                Accept: 'application/json',
                Authorization: "Bearer " + access_token,
                'content-type': 'application/json'
            }
        }).then(response => {
            setData(response.data);
        })
        .catch((error) => {
            console.log(error)
        });
    };

     const TopArtistsMediumTerm = () => {
        axios
        .get(
            TOP_ARTISTS_ENDPOINT, {
                params: {limit: 10, offset: 0, time_range: 'medium_term',},
            headers: {
                Accept: 'application/json',
                Authorization: "Bearer " + access_token,
                'content-type': 'application/json'
            }
        }).then(response => {
            setData(response.data);
        })
        .catch((error) => {
            console.log(error)
        });
    };

     const TopArtistsShortTerm = () => {
        axios
        .get(
            TOP_ARTISTS_ENDPOINT, {
                params: {limit: 10, offset: 0, time_range: 'short_term',},
            headers: {
                Accept: 'application/json',
                Authorization: "Bearer " + access_token,
                'content-type': 'application/json'
            }
        }).then(response => {
            setData(response.data);
        })
        .catch((error) => {
            console.log(error)
        });
    };

    

  

    return (
        <>
        <button class="astext" onClick={TopArtistsShortTerm}>Past 4 weeks top artists</button>   
        <button  class="astext" onClick={TopArtistsMediumTerm}>Past 6 months top artists</button>
        <button   class="astext" onClick={TopArtistsLongTerm}>Get all-time top artists</button>

        {/* {data?.items ? data.items.map((item) => <p>{item.name}</p>) : null} */}
        
        {data?.items ? data.items.map((item) =>
            <div className="container" key={item.name + '_' + item.images[1].url}>
                <div className="rank-column"><p className="rank">{data.items.indexOf(item)+1}</p></div>
                <div className="image-column"><img src= {item.images[1].url}></img></div>
                <div className="names-column"><p>{item.name}</p></div>
                </div>               
            )
        : null
        }
        
        
        
        
        
        
        
        
        
        </>
    ) 
    

};



export default GetTopArtists
