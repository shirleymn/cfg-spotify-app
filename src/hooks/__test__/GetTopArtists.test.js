import React from 'react';
import ReactDOM from 'react-dom';
import GetTopArtists from '../GetTopArtists';


import {render, cleanup, screen, waitForElement, getByTestId } from '@testing-library/react';
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event"


afterEach(cleanup)


it("renders without crashing", ()=>{
    const div = document.createElement("div");
    ReactDOM.render(<GetTopArtists></GetTopArtists>, div)
}) ;

it("matches snapshot", () =>{
    const tree = renderer.create(<GetTopArtists></GetTopArtists>).toJSON();
    expect(tree).toMatchSnapshot();

});




