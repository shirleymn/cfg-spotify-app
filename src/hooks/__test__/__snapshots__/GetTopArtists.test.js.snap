// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <button
    className="text-button"
    onClick={[Function]}
  >
    Last 4 Weeks
  </button>,
  <button
    className="text-button"
    onClick={[Function]}
  >
    Last 6 Months
  </button>,
  <button
    className="text-button"
    onClick={[Function]}
  >
    All Time
  </button>,
  <br />,
  " ",
  <br />,
]
`;
